table/tables/sys_var_tables.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tables
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/meta/autoid"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/util/types"
+)
+
+type SysVarHandle interface {
+	// GetRows is used to extract data from struct in memory
+	GetRows(ctx context.Context, cols []*table.Column) (fullRows [][]types.Datum, err error)
+}
+
+// SysVarTable stands for the fake table all its data is in the memory.
+// handle: the function to get rows
+// meta: table meta data
+// cols: column info for the table
+// @TODO this table is almost the same as the infoschema tables, but we need to use it in performance schema.
+// @TODO So we have to move it here, sometimes we need to refactor the infoschema tables to decrease the multiplicity of the codes
+type SysVarTable struct {
```

Comment:
Maybe the name `SysVarTable` is too specific, it can be used for any table that generates data on request.
How about `VirtualTable`?

And we can move add `meta` and `cols` getter to `VirtualDataSource` interface.